### PR TITLE
feat(api): add first-class operation listing filters

### PR DESCRIPTION
**Pull request by dimileeh**
Automatically opened by AWF workspace `ws_8a045867abf44287b7c32f55` (agent: `gemini`).

### Task
Strict TDD requirements:
- Start by adding focused failing tests for the requested behavior.
- Run the narrow tests and include the red failure output in the first commit body.
- Implement only the requested slice until the tests are green.
- Run the validation commands listed below before final response.
- Commit locally with a conventional commit message.
- Do not push, open PRs, switch branches, rebase, or force-push; AWF owns branch and PR lifecycle.
- Keep changes tightly scoped to the listed files/areas and avoid unrelated refactors.

PRD backing: docs/awf_prd_v2.2.md sections 14.1, 14.2, and 14.3 define operation as a first-class API resource with pagination-ready list semantics.

Harden the operation listing API.

Expected behavior:
- Add GET /v1/operations with filters: workspace_id, status, type, limit.
- Keep the existing response shape { items, next_cursor: null, has_more: false }.
- Newest-first ordering by created_at/id, consistent with events/workspaces.
- Bound limit to 1..500.
- Extend GET /v1/workspaces/{workspace_id}/operations to support optional status and type filters.
- Workspace-scoped operation list should return 404 NOT_FOUND for a missing workspace, matching other workspace subresources.
- Preserve GET /v1/operations/{operation_id}.
- Add focused tests for global list, workspace filter, status/type filters, limit validation, missing workspace, and empty result.

Validation commands to run:
uv run ruff check src/awf/api/routes/operations.py src/awf/db tests/unit/api
uv run mypy src/awf/api src/awf/db
uv run pytest tests/unit/api -q

---
Validation: 6 profile command(s) passed inside the workspace container.

### CODE REVIEW

**Comment by dimileeh** [comment]
✅ PR #25 is ready to merge at commit `5362144743`.

All 5 AWF gates are green:
1. Inline comments resolved.
2. Outside-diff comments addressed.
3. CI checks all SUCCESS or SKIPPED.
4. Mergeable.
5. Base merged into head.

AWF will not merge this PR automatically — human action required. If new commits land here, AWF will re-verify all 5 gates and re-post this message on the new head SHA.

**Comment by dimileeh** [comment]
✅ PR #25 is ready to merge at commit `6e6a4a1202`.

All 5 AWF gates are green:
1. Inline comments resolved.
2. Outside-diff comments addressed.
3. CI checks all SUCCESS or SKIPPED.
4. Mergeable.
5. Base merged into head.

AWF will not merge this PR automatically — human action required. If new commits land here, AWF will re-verify all 5 gates and re-post this message on the new head SHA.

File: src/awf/db/repositories.py [hunk 1]
```diff
@@ -230,15 +230,40 @@ async def create(
...
+        operation_type: OperationType | str | None = None,
+        limit: int = 50,
+    ) -> list[Operation]:
+        stmt = select(Operation)
+        if workspace_id is not None:
+            stmt = stmt.where(Operation.workspace_id == workspace_id)
+        if status is not None:
+            stmt = stmt.where(Operation.status == status)
+        if operation_type is not None:
+            stmt = stmt.where(Operation.type == operation_type)
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/db/repositories.py]: fixed in commit 5362144743da6b7d6f2cd86ccc1bb0519c8807fd

File: tests/unit/api/test_operations_listing.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    ws = await ws_repo.create(
+        repo_url="https://github.com/org/repo_empty",
+        branch_base="main",
+        task_title="task",
+        task_prompt="prompt",
+        agent="claude-3-sonnet",
+        test_commands=[]
+    )
+    await session.commit()
+
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply tests/unit/api/test_operations_listing.py:156]: fixed in commit bca5f3a546c8747996259f1dd9ef5264b0f9898f

File: tests/unit/api/test_operations_listing.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    # Empty global list
+    response = await client.get("/v1/operations?type=validate")
+    assert response.status_code == 200
+    assert response.json()["items"] == []
+
+    # Empty workspace list
+    response = await client.get(f"/v1/workspaces/{ws.id}/operations")
+    assert response.status_code == 200
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply tests/unit/api/test_operations_listing.py:164]: FALSE POSITIVE: tests/conftest.py provides a function-scoped in-memory SQLite engine fixture, so each test gets a fresh database and committed sample_data rows cannot leak into test_list_operations_empty.

File: src/awf/db/repositories.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/db/repositories.py]: fixed in commit cff81edae87d4c15e6dcfd0168d6cee74044ecb3

File: src/awf/db/repositories.py [hunk 4]
```diff
@@ -230,15 +230,40 @@ async def create(
...
+
+        stmt = stmt.order_by(Operation.created_at.desc(), Operation.id.desc()).limit(limit)
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/db/repositories.py:253]: fixed in commit 11619ddaad0a77eb86468ae7e2b951f489c34388

File: tests/unit/api/test_operations_listing.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    await session.commit()
+
+    response = await client.get(f"/v1/operations?workspace_id={ws.id}")
+    assert response.status_code == 200
+    items = response.json()["items"]
+    assert len(items) == 3
+    # Newest first
+    assert items[0]["id"] == op3.id
+    assert items[1]["id"] == op2.id
+    assert items[2]["id"] == op1.id
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply tests/unit/api/test_operations_listing.py:142]: fixed in commit bca5f3a546c8747996259f1dd9ef5264b0f9898f

**Comment by dimileeh** [comment]
✅ PR #25 is ready to merge at commit `df27f9a969`.

All 5 AWF gates are green:
1. Inline comments resolved.
2. Outside-diff comments addressed.
3. CI checks all SUCCESS or SKIPPED.
4. Mergeable.
5. Base merged into head.

AWF will not merge this PR automatically — human action required. If new commits land here, AWF will re-verify all 5 gates and re-post this message on the new head SHA.
